actions/kexec/v1/cmd/kexec.go

```diff
@@ -0,0 +1,77 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"golang.org/x/sys/unix"
+)
+
+// Release - this struct contains the release information populated when building kexec
+var Release struct {
+	Version string
+	Build   string
+}
+
+const mountAction = "/mountAction"
+
+var kexecCmd = &cobra.Command{
+	Use:   "kexec",
+	Short: "This is a tool for managing storage for bare-metal servers",
+	Run: func(cmd *cobra.Command, args []string) {
+
+		// Parse the environment variables that are passed into the action
+		blockDevice := os.Getenv("BLOCK_DEVICE")
+		filesystemType := os.Getenv("FS_TYPE")
+		kernelPath := os.Getenv("KERNEL_PATH")
+		initrdPath := os.Getenv("INITRD_PATH")
+		cmdLine := os.Getenv("CMD_LINE")
+
+		if blockDevice == "" {
+			log.Fatalf("No Block Device speified with Environment Variable [BLOCK_DEVICE]")
+		}
+
+		// Create the /mountAction mountpoint (no folders exist previously in scratch container)
+		err := os.Mkdir(mountAction, os.ModeDir)
+		if err != nil {
+			log.Fatalf("Error creating the action Mountpoint [%s]", mountAction)
+		}
+
+		err = syscall.Mount(blockDevice, mountAction, filesystemType, 0, "")
+		if err != nil {
+			log.Fatalf("Mounting [%s] -> [%s] error [%v]", blockDevice, mountAction, err)
+		}
```

Comment:
I think it might be easier if we have a previous step extract the kernel/initrd and place it somewhere in a `volume`'d dir and then throw the location into `ephemeral_data`. This current setup works for easy single-device setups but will start to get hairy if we start working with dm-raid or similar setups. If instead we expect for image2disk (or similar) to set us up for kexec then the kexec action is much more concise.

Reply:
I think we briefly discussed it during the stream when implementing the action.. Or I don't remember if it was when we were on zoom but

> If instead, we expect for image2disk (or similar) to set us up for kexec then the kexec action is much more concise.

The idea was to make image2disk smarter and leave kexec more concise

Reply:
yeah, the `kexec` action should focus on accessing a block device and kexecing.. we can easily no-op the mount and point to a different kernel/initrd if a different action has placed it somewhere. The first goal for this is for simple deployments and OS's that have come from a disk image.

Reply:
yep that sounds fair